.github/workflows/ci-pull.yml

```diff
@@ -1,10 +1,8 @@
 name: ci
 
 on:
+  workflow_dispatch:
   pull_request:
-  push:
-    branches:
-      - master
```

Comment:
I think we'd still want our PR checks to run on master to catch and obvious failures (e.g., PR CI passes separately for branch A and B independently, but not for the result of merging both into master)

Reply:
PR checks are more or less a subset of what this proposed CI does. We can still run it for a commit but it should not add any value

Reply:
Limited value for us, but external forks won't have access to our self-hosted runners, and I'd want them to have sufficiently functional CI out-of-the-box.

Reply:
Not quite sure I fully understand the concern but I re-enabled the workflow for commits into master branch.